source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -471,6 +473,14 @@ FilterHeadersStatus Filter::decodeHeaders(RequestHeaderMap& headers, bool end_st
   return status;
 }
 
+ProcessorState::CallbackState Filter::getCallbackStateWhenStreamBody(ProcessorState& state) const {
```

Comment:
const ProcessorState&

Actually do we need this helper function? It seems only be used in one place.

Reply:
done of the "const"
For the helper function, I am trying to put these logic in a method function to avoid grow onData() to be too big.

Reply:
This logic is pretty complicated in my opinion. i.e, the new state is StreamedBodyCallback if the original state is Idle, or StreamedBodyCallback. If the original state is HeadersCallback, then stay with that state.

I would hide such logic in a helper function than directly put it in onData().

Reply:
Are you say the if check logic below is very complicated?
```
if (state.callbackState() == ProcessorState::CallbackState::HeadersCallback) {
    return ProcessorState::CallbackState::HeadersCallback;
  } else {
    return ProcessorState::CallbackState::StreamedBodyCallback;
  }
  ```

Reply:
Yeah, it took me some time to figure out this 3 x 2 state transition, and what's the message sequence between downstream client <---->Envoy <----> Side stream server to hit those state transitions. I am actually glad it looks very simple at the end. 

If this is really a big deal for you, I can move it inside to onData() function.

Reply:
done